### ISSUE TITLE
Cannot install cargo-watch on Windows

### ISSUE THREAD

**Issue by Hoodad** [watchexec/cargo-watch#196]
Hello!

I'm running Rust 1.59 on Windows and I'm not able to install cargo-watch through "cargo install cargo-watch" or by specifying older versions like 7.8.1, or 6.0.0. 

The error I get when trying compile the latest version of cargo-watch from cargo install is a lot of lines like the following 

`          cargo_watch.cargo_watch.fe3b8ab8-cgu.0.rcgu.o : error LNK2019: unresolved external symbol _ZN94_$LT$$RF$alloc..vec..Vec$LT$T$C$A$GT$$u20$as$u20$core..iter..traits..collect..IntoIterator$GT$9into_iter17h8ddd951a7ac859beE referenced in function _ZN11cargo_watch4main17hf77abfe770f08e97E
          cargo_watch.cargo_watch.fe3b8ab8-cgu.0.rcgu.o : error LNK2019: unresolved external symbol _ZN3std2fs12canonicalize17hc8a2ff6bab86b373E referenced in function _ZN11cargo_watch4main17hf77abfe770f08e97E
          cargo_watch.cargo_watch.fe3b8ab8-cgu.0.rcgu.o : error LNK2019: unresolved external symbol _ZN3std4path4Path3new17h8fabc76080651543E referenced in function _ZN11cargo_watch4main17hf77abfe770f08e97E
          cargo_watch.cargo_watch.fe3b8ab8-cgu.0.rcgu.o : error LNK2019: unresolved external symbol _ZN72_$LT$alloc..vec..Vec$LT$T$C$A$GT$$u20$as$u20$core..ops..deref..Deref$GT$5deref17h7bc95302f54b69d7E referenced in function _ZN11cargo_watch4main17hf77abfe770f08e97E`

And the last line in the error is
"\cargo-watch\target\release\deps\libnotify_rust-5f7e0bc1b07b293d.rlib : fatal error LNK1107: invalid or corrupt file: cannot read at 0x42496"

If I clone the repo I can build cargo-watch if I build debug version. But the release version grants me the same error as through cargo install. 

Not sure if I have missed something in the installation guides or not but I cannot proceed from here, any help is greatly appreciated!

## Versions
rustc 1.59.0 (9d1b2106e 2022-02-23)
Windows 11 Enterprise, Version 10.0.22000 Build 22000

**Comment by passcod** [comment]
Looks like your Rust installation is corrupted. Try reinstalling it, or deleting and installing anew your toolchain via rustup. This is unlikely to be a cargo-watch bug.

That said, you can install cargo-watch from binary: https://watchexec.github.io/downloads/cargo-watch/ or via some package managers like Chocolatey or with [binstall](https://github.com/ryankurte/cargo-binstall).

**Comment by Hoodad** [comment]
Thanks for getting back!

I didn't mention in my previous post but I had tried compiling using nightly (1.62) and now after your recommendation I have uninstalled and installed rust 1.60 and no mater what I do I keep getting the same error as mentioned before.

How do I use binstall to install cargo-watch? I have tried the following

Installing using the following commands
* cargo install cargo-binstall (ok)
* cargo binstall cargo-watch (ok)

Then when I run "cargo watch" I get  

```
cargo watch --echo Hello World
error: no such subcommand: `watch`

        Did you mean `fetch`?
```

So I'm at a loss of how to proceed. 

Either way to get cargo-watch installed is valid for me, I just wanna use the software :D

**Comment by passcod** [comment]
That's odd, if `cargo-binstall` installed correctly. Your `PATH` may be incorrect? It should include the `cargo/bin` directory. Or the way that cargo-binstall installed it may be wrong...

If you call `cargo-watch` directly, does that work?

(Also note `--echo` isn't a cargo watch option, but one thing at a time)

**Comment by Hoodad** [comment]
I checked earlier and the PATH is good, clippy and fmt all work as expected.

However thanks to you tip about the `cargo/bin` I checked out the folder and noticed that cargo-watch was using a symlink to `cargo-watch-8.1.1.exe` (where as clippy and fmt was not). By removing the symlink and renaming the exe to just `cargo-watch.exe` things started to work 👍 

Not sure if symlinks have changed in Win 11 or not but that seemed to have been the issues as binstall worked just fine :) 

Let me know if there is anything you want me to try if you want to keep on digging. Otherwise I consider my issue here closed, many thanks!
